### ISSUE TITLE
Dedicated server will not download large workshop addons

### ISSUE THREAD

**Issue by lopsided98** [Facepunch/garrysmod-issues#1520]
This bug has been known for some time, but I was unable to find an existing report of it here; let me known if this has already been reported.

When downloading a collection of addons to a dedicated server from the workshop, the server gives up after a certain amount of time. This occurs with most addons, except for very small ones.

Here some links to discussions about the bug:
http://steamcommunity.com/workshop/discussions/-1/540744937178248834/?appid=4000
http://forums.srcds.com/viewtopic/19594

Here is an example from my server's log:

```
Mounting Addon 'M9K Heavy Weapons' (128091208)
Downloading...
  1%
  2%
  2%
  3%
  4%
  5%
  6%
  7%
  8%
  9%
Download Failed!
Mounting Addon 'M9K Small Arms pack' (128093075)
Downloading...
  1%
  1%
  2%
  2%
  3%
  4%
  5%
  5%
  6%
  7%
  8%
  8%
  9%
  10%
  10%
Download Failed!
Mounting Addon 'M9K Specialties' (144982052)
Downloading...
  1%
  2%
  3%
  4%
  5%
  5%
  6%
  7%
  8%
  9%
  10%
  11%
  12%
  13%
  14%
Download Failed!
```

This bug can be worked around by manually copying the addons from a client install, but it would be nice to have a actual fix to this annoying problem.

**Comment by ajwharton** [comment]
I would be shocked if the scrds server was buffering this to memory and not just dumping it to /tmp someplace. That said, this stupid 64M limit is starting to bother me. This would seem to be an extremely straightforward fix and the failure to deal with it is irritating. As someone said in ITT, this isn't a problem with the client, so it is clearly an arbitrary number somewhere in the server code.

**Comment by PatPeter** [comment]
I literally stopped adding new content to my server from the workshop because I absolutely despise having to download the file in my client, rename it, and then upload it to the server.

**Comment by lopsided98** [comment]
I was just wondering if there are plans to fix this bug, as it was added to the mounting milestone.

On the other hand, I came up with my own workaround, available [here](https://gist.github.com/lopsided98/79c38067c701a3281adc). It automatically copies and renames addons from a client install to a server over ssh. It only works on Linux (I guess it could work on Windows with some work), and my system's paths are hardcoded in, but it still might be helpful if you know that your doing.
